client/transaction-pool/src/testing/pool.rs

```diff
@@ -488,7 +488,7 @@ fn fork_aware_finalization() {
 
 	// block e1
 	{
-		let header = pool.api.push_block(5, vec![from_dave]);
+		let header = pool.api.push_block(5, vec![from_dave, from_bob]);
```

Comment:
`from_bob` should already be in the pool when `D2` is retracted, no? Can you add some assertions about the pool status after each block (i.e. how many transactions we expect to be in the pool).

Reply:
Yo @tomusdrw, this should be ready now